libs/designsystem/src/lib/components/empty-state/empty-state.component.ts

```diff
@@ -1,13 +1,58 @@
-import { Component, Input } from '@angular/core';
+import {
+  AfterContentInit,
+  Component,
+  ContentChildren,
+  Input,
+  OnDestroy,
+  QueryList,
+} from '@angular/core';
+import { Subject } from 'rxjs';
+import { delay, takeUntil } from 'rxjs/operators';
+
+import { ButtonComponent } from '../button/button.component';
 
 @Component({
   selector: 'kirby-empty-state',
   templateUrl: './empty-state.component.html',
   styleUrls: ['./empty-state.component.scss'],
 })
-export class EmptyStateComponent {
+export class EmptyStateComponent implements AfterContentInit, OnDestroy {
   @Input() iconName: string;
   @Input() customIconName: string;
   @Input() title: string;
   @Input() subtitle: string;
+
+  @ContentChildren(ButtonComponent)
+  private slottedButtons: QueryList<ButtonComponent>;
+
+  private ngUnsubscribe$ = new Subject<void>();
+
+  ngAfterContentInit() {
+    this.enforceAttentionLevelRules();
+
+    this.slottedButtons.changes
+      .pipe(takeUntil(this.ngUnsubscribe$), delay(0))
+      .subscribe(() => this.enforceAttentionLevelRules());
+  }
+
+  ngOnDestroy() {
+    this.ngUnsubscribe$.next();
+    this.ngUnsubscribe$.complete();
+  }
+
```

Comment:
No need to unsubscribe as this is handled by Angular when the component gets destroyed
```suggestion
```

Reply:
Should be resolved by 867ba3e